src/elm/Session/LoggedIn.elm

```diff
@@ -293,8 +308,79 @@ viewFeedback status message =
         ]
 
 
-viewHelper : (Msg -> msg) -> Page -> Profile.Model -> Model -> Html msg -> Html msg
-viewHelper thisMsg page profile_ ({ shared } as model) content =
+viewHelper : (Msg -> pageMsg) -> Page -> Profile.Model -> Model -> Html pageMsg -> Html pageMsg
+viewHelper pageMsg page profile_ ({ shared } as model) content =
+    let
+        { t } =
+            shared.translators
+    in
+    div
+        [ class "min-h-screen flex flex-col" ]
+        ([ div [ class "bg-white" ]
+            [ div [ class "container mx-auto" ]
+                [ viewHeader model profile_
+                    |> Html.map pageMsg
+                , if Search.isActive model.searchModel then
+                    text ""
+
+                  else
+                    viewMainMenu page model |> Html.map pageMsg
+                ]
+            , case model.feedback of
+                Show status message ->
+                    viewFeedback status message |> Html.map pageMsg
+
+                Hidden ->
+                    text ""
+            ]
+         ]
+            ++ (let
+                    viewClaimWithProofs action proof =
+                        [ Action.viewClaimWithProofs proof shared.translators (isAuth model) action
+                            |> Html.map (GotActionMsg >> pageMsg)
+                        ]
+                in
+                case ( Search.isActive model.searchModel, model.claimingAction.status ) of
+                    ( True, _ ) ->
+                        [ Search.viewSearchBody
+                            shared.translators
+                            model.selectedCommunity
+                            model.date
+                            (GotSearchMsg >> pageMsg)
+                            (GotActionMsg >> pageMsg)
+                            model.searchModel
```

Comment:
I managed to move `veiwSearchBody` into the `Search` module and also changed the way we check what to show on the page from `if`s to `case` with tuples.